homeassistant/components/abode/__init__.py

```diff
@@ -302,7 +302,7 @@ def should_poll(self):
 
     async def async_will_remove_from_hass(self):
         """Unsubscribe from Abode connection status updates."""
-        self.hass.async_add_job(
+        self.hass.async_add_executor_job(
```

Comment:
Don't we want to await the task so that it's finished before we continue?

Reply:
You mean adding `await` to this method? Yeah, that would make sense since `async_add_executor_job` is a coroutine.

However, maybe I'm misunderstanding your question. By not adding `await`, isn't it actually making this a blocking method (i.e. we do wait until its complete before moving on)? I'm not saying this was intentional or desired, more just for my learning since I thought making something awaitable in asyncio allows other coroutines in the event loop to execute while it's waiting.

I'll make another PR now to fix this though. Thanks again!

Reply:
`hass.async_add_executor_job` is a callback. It schedules work on the executor thread pool and returns a task with the result.

https://github.com/home-assistant/core/blob/d2e70eb967b962a28bfdd43c9e5a7452919789b1/homeassistant/core.py#L338-L349

By awaiting the returned task we yield control to the event loop and suspend the coroutine where we currently are, ie `async_added_to_hass`. When the task is finished we resume the coroutine and continue its execution.

If we don't await the task it will run in the background and eventually finish but we'll not be notified and we will continue with the execution of `async_added_to_hass` before the task is done.